src/components/Compare/CompareTable.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState } from 'react';
+import { TableBody } from '@material-ui/core';
+import {
+  Wrapper,
+  StyledTable,
+  ArrowContainer,
+  TableHeadContainer,
+  Row,
+  Cell,
+  Footer,
+  ViewAllLink,
+  Header,
+} from 'components/Compare/Compare.style';
+import CompareTableRow from 'components/Compare/CompareTableRow';
+import HeaderCell from 'components/Compare/HeaderCell';
+import { getLocationNames, Location } from 'common/locations';
+import {
+  stateSummary,
+  countySummary,
+  LocationSummary,
+} from 'common/location_summaries';
+import { Metric } from 'common/metric';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { COLOR_MAP } from 'common/colors';
+import { fail } from 'common/utils';
+
+const locations: any = getLocationNames();
+
+export interface SummaryForCompare {
+  locationInfo: Location;
+  metricsInfo: LocationSummary;
+}
+
+const CompareTable = (props: {
+  stateId?: string;
+  stateName?: string;
+  county?: any | null;
+  setShowModal: any;
+  isModal?: Boolean;
+  locationsViewable?: number;
+  isLocationPage?: Boolean;
+  isHomepage?: Boolean;
+}) => {
+  const [sorter, setSorter] = useState(5);
+  const [sortDescending, setSortDescending] = useState(true);
+
+  const orderedMetrics = [
+    Metric.CASE_DENSITY,
+    Metric.CASE_GROWTH_RATE,
+    Metric.POSITIVE_TESTS,
+    Metric.HOSPITAL_USAGE,
+    Metric.CONTACT_TRACING,
+  ];
+
+  const statesArr = locations
+    .filter((location: Location) => !location.county)
+    .map((stateInfo: Location) => {
+      if (stateInfo.state_code) {
+        return {
+          locationInfo: stateInfo,
+          metricsInfo: stateSummary(stateInfo.state_code),
+        };
+      } else {
+        fail('No state code');
+      }
+    });
+
+  const countiesArr = locations
+    .filter(
+      (location: Location) =>
+        location.county && location.state_code === props.stateId,
+    )
+    .map((countyInfo: Location) => {
+      if (countyInfo.full_fips_code) {
+        return {
+          locationInfo: countyInfo,
+          metricsInfo: countySummary(countyInfo.full_fips_code),
+        };
+      } else {
+        fail('No county fips');
+      }
+    });
+
+  const locationsArr = props.isHomepage ? statesArr : countiesArr;
+
+  locationsArr.sort((a: any, b: any) => {
+    if (a.metricsInfo.metrics && b.metricsInfo.metrics) {
+      if (
+        a.metricsInfo.metrics[sorter].value >
+        b.metricsInfo.metrics[sorter].value
+      ) {
+        return 1;
+      } else {
+        return -1;
+      }
+    }
+  });
```

Comment:
Consider using something simpler to sort, maybe use `lodash`'s `_.sortBy`. Example:

```tsx
const locations = [
  { name: 'AZ', rt: 1.3, contactTracing: 0.34 },
  { name: 'AL', rt: 1.8, contactTracing: 0.44 },
  { name: 'NY', rt: 1.6, contactTracing: 0.55 },
  { name: 'CA', rt: 1.4, contactTracing: 0.67 },
  // ...
]
const [sortingProp, setSortingProp] = useState('name');
const sorted = sortingProp === 'contactTracing' 
  ? _.sortBy(locations, 'contactTracing').reverse() 
  : _.sortBy(locations, sortingProp);
// render rows...
```